### PR TITLE
Replaces the Hyena with the Derecho

### PR DESCRIPTION
**Pull request by Apogee-dev**
## About The Pull Request
derecho
derecho_areas
derecho_ingame

Replaces the Hyena with the Derecho-class Auxiliary Wrecker, a mid-sized New Gorlex salvage ship. Comes with more advanced engineering gear than usual and a moderate array of weaponry including two Asp pistols and one Bulldog shotgun. Features a crew of six players:
- Captain: You know what this is.
- Foreman: Direct the efforts of the Wreckers and the Mechanic.
- Wrecker x2: Explore, mine, and most especially salvage wrecks for materials and valuables. Comes with a limited number of advanced salvage supplies including power tools and a plasma cutter. One wrecker can wear a toolbelt, the other gets a mining webbing.
- Mechanic: Maintain, repair, and improve the ship.
- Initiate x2: The starter role. Keep the ship clean, your crewmates fed, and in general try to be useful while learning the ropes of the faction and/or the game.

## Why It's Good For The Game

NGR needs more ships and the Hyena's been slated for replacement for a while now due to its outdated and profoundly crusty design.

## Changelog

:cl:
add: Replaced NGR Hyena with the NGR Derecho-class salvage ship
/:cl:

### CODE REVIEW

**Review by Gristlebee — CHANGES_REQUESTED** [review]
- Please add a screenshot of the ship ingame, and with areas when you get a chance.
- Repaths broke a bunch of stuff unfortunately, that will need to be fixed.
image
- use the actual window structures instead of the spawners please
image
- these conveyers dont have a switch linked to them.
image
- this button here by the starboard engines has the wrong ID
image
- same for this button on the bridge
image
image
- the syndicate viper has a guncase coded in, (/obj/item/storage/guncase/pistol/a357) you could use that instead of an inherit case.
image
- the fax, holopad and photocopier technically dont need wiring, but if it's an aesthetic choice, thats fair enough.
image
- there doesnt appear to be a button to operate the cargo bay doors or holofield as far as I can tell.
image
- The storage bay doors could use a button on the hangar side so you don't need to loop around to open it.
image
- In terms of gear for the wreckers:
  -  I think an angle grinder would be good for the ship as basic salvage tools and for whichever wrecker doesn't get the cutter. 
  - Welding goggles/NGR mesons/eye protection being mapped in would be good for the wreckers using the cutter so they don't go blind. 
  - Survival knives would be good to have  as they're typical for most ships.
  - Mining scanners would be nice , though I suppose it's fine for them to be without depending how specialized you want them to be for shipbreaking.

**Comment by Apogee-dev** [comment]
- The wires in the command office are for electrified windows (and the APC behind the photocopier). 
- I’ve received so many contrary requests about windows and spawners at this point that I have no idea what to go with, but sure, structures it is.
- Most of the missing bits are genuine oversights or the result of working on this before certain PRs had merged, will fix those this weekend.
- The repaths (balaclavas excluded) are actually mostly types that were added in the kali andhi PR. As soon as that merges they’ll be fine.

As for wrecker gear…
- No on the grinder. The wreckers get two sets of specialized gear- one wrecker gets a toolbelt and power tools, the other gets a mining belt and plasma cutter. The intent is for one to focus on heavy jobs and the other on dismantling the more delicate stuff (and for either one to haul material if they run out of stuff to do)
- Wrecker space helmets provide eye protection.
- Survival knives will be added.
- No on mining scanners- they barely even get digging and ore handling equipment to begin with.

**Review by Gristlebee — CHANGES_REQUESTED** [review]
- There's a stray fire cabinet here on the bridge. 
image
image
- There's no publicly accessible buttons to operate the hangar doors for the crew to use from what I can see, only the ones in the foreman's office behind locked doors.
image
- Bit of a nitpick, but I think the hangar storage shutter button would look better if the tank dispenser wasn't in front of it. I know it doesn't mechanically block the button, but it feels a bit odd to me. Try moving the dispenser somewhere else and see how you feel about it. Or maybe move it one to the left where the ore box currently is and put the ore box with the rest of the cargo.
image

**Comment by Apogee-dev** [comment]
The hangar door button being only in the foreman's office is intentional. Since this ship has an officer in command of the cargo section and the wreckers, he gets control over hangar access. The tank dispenser isn't moving either, I tried a few spots and liked none of them. It doesn't even visually block the button either since the tank rack doesn't stretch floor to ceiling.

**Comment by Gristlebee** [comment]
> The hangar door button being only in the foreman's office is intentional. Since this ship has an officer in command of the cargo section and the wreckers, he gets control over hangar access.

I can understand where you're coming from, but I don't agree with this.  While it is flavorful, I don't think it would be enjoyable from a gameplay standpoint. Considering how often you need to open or close the hangar doors throughout the course of a round, restricting a basic function to the operation of a ship is effectively a lot of busywork and an artificial bottleneck.
Another problem is how finnicky holofields can be, any issues with the hangar can't be resolved until the foreman gets to it. 

The foreman is going to be stuck to their chair on door duty with how load bearing it is to the operation of the ship. While there is an argument to be made that command staff should be staying on the ship, pressing a button repeatedly is not good gameplay.

There's also the issue you are not always guaranteed a foreman joining your ship. The Captain can do it of course, but that's another set of duties artificially created by the single set of hangar controls on top of the normal responsibilities.

In short, I don't think it will meaningfully improve the experience of the ship. Though, if you get another maptainer's approval for a single set controls for the hangar door, I'll be alright with it. Otherwise, I'd prefer you add a crew accessible set of controls.

**Comment by Erikafox** [comment]
> The hangar door button being only in the foreman's office is intentional. Since this ship has an officer in command of the cargo section and the wreckers, he gets control over hangar access. The tank dispenser isn't moving either, I tried a few spots and liked none of them. It doesn't even visually block the button either since the tank rack doesn't stretch floor to ceiling.

I did this on the bolide and it sucked major dick

**Comment by Apogee-dev** [comment]
You aren’t guaranteed a foreman every round but you ARE guaranteed a captain every round, who will always be on the bridge right around the corner every time the ship lands or takes off, and will probably pass through the foreman’s office to get to cargo since it’s the fastest route. The door also only needs to be opened or closed when the ship is landing or departing- times when the foreman should probably be in his office anyway. Having to open or close the bay doesn’t chain him  to his office, he’s supposed to just go there when landing or leaving and join the Wreckers when heading out. That’s one reason it’s directly accessible from cargo. I just don’t want Random Initiate opening the doors and fucking off the instant the ship lands. The Kali’s cargo bay controls are up at the Wrecker’s balcony similarly, the difference is just a lack of a restricted access door for them on that ship since the single Wrecker doesn’t really justify a room of their own.

**Comment by Erikafox** [comment]
Ma'am I genuinely tried this and it was just... Not fun to play around.

At the least put an access locked (sec/mining) secondary button inside the bay somewhere

**Comment by Apogee-dev** [comment]
I would if I had literally any wall space left in cargo. Ough. I’ll figure something out.

**Comment by Gristlebee** [comment]
- So with the security and mining access on the button in the hangar. The issue then is the Foreman doesn't have either of those accesses so they can't use that button, which doesn't make sense if the intent is that they're supposed to be in charge of the hangar.
image
image
I still don't agree with restricting access to the hangar doors at all either way. I'd prefer if you made it open access. If an initiate runs off on their own without the permission of the foreman or anyone else, that can be something handled IC or administratively if it's really egregious. If you're still set on having the restricted/limited access, I'm alright if you ask another maptainer for approval on it and I'll be okay with it if they give their assent on it.. But right now it's a no from me.

**Comment by Erikafox** [comment]
image
foreman needs a different belt this looks like a diaper on them

**Review by Gristlebee — CHANGES_REQUESTED** [review]
Feedback after testing on live:
---

- These windows are missing grilles.
image
- The workshop door requires both engineering and mining access rather than one of them. Only the captain is able to open them right now.
image
- Still spawners here under the bridge window at the bottom
image
- Cryo has no control console.
image
- The foreman playing found it annoying how their office was a trafficked area, and the Captain had to kept shuffling past them to get by due to the layout of the doors
image
Maybe shuffle the SSU and rack over to the right, and align the doors vertically  so you don't need to move past the foreman to get by.
image
- A complaint players had was how far hangar door button in the bay was from the blast doors. Though it is understandable since the cargo bay is lacking wall space like you said before. Maybe take down one of the signs by the airlock?

**Comment by Jedi-Toothpaste** [comment]
Hi, the smeltery console also starts unlinked, this is a minor fix but it would be nice
image

EDIT: the semlter also exports to the wrong side, but I imagine this was a byproduct of fixing it inround
image

**Comment by fighterslam** [comment]
image
 you are missing a grille on this window

**Review by Gristlebee — CHANGES_REQUESTED** [review]
Platforms in the cargo bay will need to be adjusted since #3745 fixed them due to their side states being backwards.

Cryo room still has no console.

Once that's done, and the smelter is fixed, I believe this should be good to merge.

**Comment by EPSLYN** [comment]
Just tested this out on Round 2862, as captain for its first test merge. I've got a few pros and cons for it personally. It reminds me a lot of the Vaquero, honestly.

**PROS:**
- A LOT more spacious than the Hyena. 
- A lot more storage--considering it's a salvage ship, that makes sense.
- Nice smelting room.
- Much nicer stock of gear. Weapons are good enough for two wreckers and the mechanic if they have to get out and fight. 

**CONS:**
- Mechanic commented that the atmos was very small and compact. I agree, and feel that if something bad happened in there, the ship would be rather fucked as to how close it is with the engines. 
- No real armory room. I liked that the Hyena had it, and that it served as a good way to get from one side to the other quickly if left open. The guns are stored in the bridge with the ammo, which is a very weird change from the Hyena, whose armory had armor, and the guns in it too, as well as space to keep them.
- VERY long to walk inbetween rooms from one side of the ship to other. Now that isn't bad, but there is quite a lot of doors in the way to get to and fro in the ship.

**Review by Gristlebee — APPROVED** [review]
